contracts/adapters/swap/astroport/src/contract.rs

```diff
@@ -171,22 +157,40 @@ fn execute_swap(
         return Err(ContractError::Unauthorized);
     }
 
+    // reset the pre swap out asset amount
+    PRE_SWAP_OUT_ASSET_AMOUNT.save(deps.storage, &Uint128::new(0))?;
```

Comment:
I'm pretty sure in theory this isn't needed, but might be good to be safe

Reply:
Why in theory is this not needed? Just because we don't expect coinhall to ever give us routes that touch the same denom out as the offer asset?

Reply:
I believe it doesn't query the balance because it does it later, but let me recheck the logic

Reply:
Ok now that I'm looking at the code again, it's not querying the balance in `execute_swap` because it queries the balance in `execute_astroport_pool_swap`.

And this is why I said that it's not needed in theory, because whatever value gets saved here, gets overwritten anyways.